src/java/picard/vcf/CallingMetricAccumulator.java

```diff
@@ -73,34 +94,34 @@ public String getPartition(final CollectVariantCallingMetrics.VariantCallingDeta
      * then returned.
      */
     private final CollectionUtil.DefaultingMap<String, VariantCallingDetailMetrics> sampleMetricsMap =
-            new CollectionUtil.DefaultingMap<String, VariantCallingDetailMetrics>(
-                    new CollectionUtil.DefaultingMap.Factory<VariantCallingDetailMetrics, String>() {
-                        @Override
-                        public VariantCallingDetailMetrics make(final String sampleName) {
-                            final VariantCallingDetailMetrics detail = new VariantCallingDetailMetrics();
-                            detail.SAMPLE_ALIAS = sampleName;
-                            return detail;
-                        }
+            new CollectionUtil.DefaultingMap<>(
+                    sampleName -> {
+                        final VariantCallingDetailMetrics detail = new VariantCallingDetailMetrics();
+                        detail.SAMPLE_ALIAS = sampleName;
+                        return detail;
                     }, true);
 
     public CallingMetricAccumulator(final DbSnpBitSetUtil.DbSnpBitSets dbsnp) {
         this.dbsnp = dbsnp;
     }
 
+    public void setup(final VCFHeader vcfHeader){
+        //noop.
+    }
+
     /** Incorporates the provided variant's data into the metric analysis. */
     @Override
     public void accumulate(final VariantContext vc) {
-        progress.record(vc.getChr(), vc.getStart());
+        progress.record(vc.getContig(), vc.getStart());
         if (!isVariantExcluded(vc)) {
             final String singletonSample = getSingletonSample(vc);
             updateSummaryMetric(summaryMetric, null, vc, singletonSample != null); // The summary metric has no genotype.
-            for (final String sampleName : vc.getSampleNames()) {
-                // Skip homozygous reference calls.
-                if (!vc.getGenotype(sampleName).isHomRef()) {
-                    updateDetailMetric(sampleMetricsMap.get(sampleName), vc.getGenotype(sampleName), vc,
-                            sampleName.equals(singletonSample));
-                }
-            }
+
+            vc.getSampleNames().stream()
+                    .filter(sampleName -> !vc.getGenotype(sampleName).isHomRef())
```

Comment:
I don't know much about the streaming operators, but are you filtering here if the genotype is NOT hom ref?